CVLByExample/ERC20Assembly/ERC20UtilsSummary.spec

```diff
@@ -0,0 +1,110 @@
+import "ETH.spec";
+
+using DummyERC20A as erc20A;
+using DummyERC20B as erc20B;
+using DummyERC20C as erc20C;
+
+methods {
+    function ERC20Utils.getBalance(address token, address account) internal returns (uint256) with (env e) => dispatchBalanceOf(e, token, account);
+    function ERC20Utils.safeTransfer(address token, address recipient, uint256 amount) internal returns (bool) with (env e) => dispatchTransfer(e, token, recipient, amount);
+    function ERC20Utils.safeTransferFrom(address token, address sender, address recipient, uint256 amount) internal returns (bool) with (env e) => dispatchTransferFrom(e, token, sender, recipient, amount);
+    function ERC20Utils.approve(address token, address to) internal with (env e) => dispatchApprove(e, token, to);
+    function ERC20Utils.permit(address token, bytes calldata data) internal with (env e) => dispatchPermit(e, token, data);
+    function ERC20Utils.isETH(address token, uint256 amount) internal returns (uint256) with (env e) => isETHSummary(token, amount, e.msg.value);
+}
+
+function isETHSummary(address token, uint256 amount, uint256 msgvalue) returns uint256 {
+    if (token == ETH()) {
+        require msgvalue == amount;
+        return 1;
+    } else {
+        require msgvalue == 0;
+    }
+    return 0;
+}
+
+function dispatchBalanceOf(env e, address token, address account) returns uint256 {
+    env ed; /// Dispatch environment
+    require ed.msg.sender == currentContract;
+    require ed.block.timestamp == e.block.timestamp; 
+    require ed.msg.value == 0;
+    if (token == erc20A) {
+        return erc20A.balanceOf(ed, account);
+    } else if (token == erc20B) {
+        return erc20B.balanceOf(ed, account);
+    } else if(token == erc20C) {
+        return erc20C.balanceOf(ed, account);
+    } else if (token == ETH()) {
+        return nativeBalances[account];
+    } else {
+        require false; // optimistic
+    }
+    return 0;
+}
+
+function dispatchTransfer(env e, address token, address recipient, uint256 amount) returns bool {
+    env ed; /// Dispatch environment
+    require ed.msg.sender == currentContract;
```

Comment:
do we have `executingContract` in this point? otherwise `require ed.msg.sender == currentContract;` can be a wrong assumptions

Reply:
That's true (even though here it's the only option currentContract = executingContract).
Do we have a keyword then?